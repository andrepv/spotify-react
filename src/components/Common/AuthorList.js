import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import { ARTIST } from "constants/RouteConstants";

export default class AuthorList extends Component {
  shouldComponentUpdate(nextProps) {
    return this.props.authors !== nextProps.authors;
  }

  render() {
    const {authors, handler} = this.props;
    return authors.map((author, index) => {
      return (
        <React.Fragment key={author.id}>
          <Link to={`${ARTIST}/${author.id}`}>
            <span onClick={handler}>
              {author.name}
            </span>
          </Link>
          {index + 1 !== authors.length && ", "}
        </React.Fragment>
      );
    });
  }
}

AuthorList.propTypes = {
  authors: PropTypes.array.isRequired,
  handler: PropTypes.func,
};