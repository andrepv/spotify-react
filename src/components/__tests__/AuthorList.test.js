import React from "react";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import AuthorList from "components/Common/AuthorList";

describe("<AuthorList />", () => {
  it("renders correctly", () => {
    const props = {
      authors: [
        {name: "John Doe", id: "1"},
        {name: "Mike", id: "2"},
      ],
      handler: jest.fn(),
    };
    const component = renderer.create(
      <Router><AuthorList {...props}/></Router>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it("check the component with the single author", () => {
    const props = {
      authors: [{name: "John Doe", id: "1"}],
      handler: jest.fn(),
    };
    const wrapper = shallow(<Router><AuthorList {...props} /></Router>);
    expect(wrapper.html()).toEqual(
      "<a href=\"/artist/1\"><span>John Doe</span></a>"
    );
  });
});