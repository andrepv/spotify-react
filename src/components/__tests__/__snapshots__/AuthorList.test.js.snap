// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorList /> renders correctly 1`] = `
Array [
  <a
    href="/artist/1"
    onClick={[Function]}
  >
    <span
      onClick={[MockFunction]}
    >
      John Doe
    </span>
  </a>,
  ", ",
  <a
    href="/artist/2"
    onClick={[Function]}
  >
    <span
      onClick={[MockFunction]}
    >
      Mike
    </span>
  </a>,
]
`;
